Objects/typevarobject.c

```diff
@@ -0,0 +1,1313 @@
+};
+
+PyObject *_Py_make_typevar(const char *name, PyObject *bound_or_constraints) {
+    if (bound_or_constraints != NULL && PyTuple_CheckExact(bound_or_constraints)) {
+        return (PyObject *)typevarobject_alloc(name, NULL, bound_or_constraints, false, false, true);
+    } else {
+        return (PyObject *)typevarobject_alloc(name, bound_or_constraints, NULL, false, false, true);
+    }
+}
+
+PyObject *_Py_make_paramspec(const char *name) {
+    return (PyObject *)paramspecobject_alloc(name, NULL, false, false, true);
+}
+
+PyObject *_Py_make_typevartuple(const char *name) {
+    return (PyObject *)typevartupleobject_alloc(name);
+}
+
+static void
+typealias_dealloc(PyObject *self)
+{
+    typealiasobject *ta = (typealiasobject *)self;
+    free((void *)ta->name);
+    Py_XDECREF(ta->type_params);
+    Py_XDECREF(ta->compute_value);
+    Py_XDECREF(ta->value);
+    Py_TYPE(self)->tp_free(self);
+}
+
+static PyObject *
+typealias_repr(PyObject *self)
+{
+    typealiasobject *ta = (typealiasobject *)self;
+    return PyUnicode_FromFormat("%s", ta->name);
+}
+
+static PyMemberDef typealias_members[] = {
+    {"__name__", T_STRING, offsetof(typealiasobject, name), READONLY},
+    {"__type_params__", T_OBJECT, offsetof(typealiasobject, type_params), READONLY},
+    {0}
+};
+
+static typealiasobject *
+typealias_alloc(const char *name, PyObject *type_params, PyObject *compute_value)
+{
+    typealiasobject *ta = PyObject_GC_New(typealiasobject, &_PyTypeAlias_Type);
+    if (ta == NULL) {
+        return NULL;
+    }
+    ta->name = strdup(name);
+    if (ta->name == NULL) {
+        Py_DECREF(ta);
+        return NULL;
+    }
+    ta->type_params = Py_XNewRef(type_params);
+    ta->compute_value = Py_NewRef(compute_value);
+    ta->value = NULL;
+    PyObject_GC_Track(ta);
+    return ta;
+}
+
+static int typealias_traverse(typealiasobject *self, visitproc visit, void *arg)
+{
+    Py_VISIT(self->type_params);
+    Py_VISIT(self->compute_value);
+    Py_VISIT(self->value);
+    return 0;
+}
+
+/*[clinic input]
+typealias.__reduce__ as typealias_reduce
+
+[clinic start generated code]*/
+
+static PyObject *
+typealias_reduce_impl(typealiasobject *self)
+/*[clinic end generated code: output=913724f92ad3b39b input=4f06fbd9472ec0f1]*/
+{
+    return PyUnicode_FromString(self->name);
+}
+
+static PyMethodDef typealias_methods[] = {
+    TYPEALIAS_REDUCE_METHODDEF
+    {0}
+};
+
+PyDoc_STRVAR(typealias_doc,
+"Type alias.\n\
+\n\
+Type aliases are created through the type statement:\n\
+\n\
+  type Alias = int\n\
+");
+
+PyTypeObject _PyTypeAlias_Type = {
+    PyVarObject_HEAD_INIT(&PyType_Type, 0)
+    .tp_name = "TypeAlias",
+    .tp_basicsize = sizeof(typealiasobject),
+    .tp_dealloc = typealias_dealloc,
+    .tp_repr = typealias_repr,
+    .tp_flags = Py_TPFLAGS_DEFAULT | Py_TPFLAGS_IMMUTABLETYPE,
+    .tp_members = typealias_members,
+    .tp_methods = typealias_methods,
+    .tp_doc = typealias_doc,
+    .tp_traverse = typealias_traverse,
+};
```

Comment:
`tp_new` and `tp_free` not set. And `tp_free` is called by `tp_dealloc = typevarobject_dealloc`

Reply:
I'm not sure we need tp_new; there's no need to create these things from Python. However, test_type_aliases is segfaulting due to some GC-related issue; I'm trying to figure that out right now.